python/python/lance/dataset.py

```diff
@@ -227,9 +227,19 @@ def versions(self):
         """
         versions = self._ds.versions()
         for v in versions:
-            v["timestamp"] = datetime.fromtimestamp(v["timestamp"])
+            ts_nanos = v["timestamp"]
+            v["timestamp"] = datetime.fromtimestamp(ts_nanos // 1e9) + timedelta(
+                microseconds=(ts_nanos % 1e9) // 1e3
+            )  # NOTE: python datetime supports only microsecond precision
```

Comment:
can you add a TODO here? At some point we could add a Version class to python as well that holds the Rust nanosecond timestamp and then has methods to expose it as python datetime or something that can be compared to time.time_ns(). But not worth investing in that right now

Reply:
Done.